### ISSUE TITLE
Custom alert template example fails to send notification to Discord; produces error `Post \"\": unsupported protocol scheme \"\"`

### ISSUE THREAD

**Issue by shiinee** [grafana/grafana#43826]
**What happened**:

We defined a simple alert template called `basic` which looks like this:

```
{{ define "alert" }}
  {{ .Annotations.summary }} 
{{ end }}

{{ define "message" }}
  {{ if gt (len .Alerts.Firing) 0 }}
    {{ len .Alerts.Firing }} firing:
    {{ range .Alerts.Firing }} {{ template "alert" .}} {{ end }}
  {{ end }}
{{ end }}
```

Under the Discord contact point, we changed the message content from `{{ template "default.message" . }}` to `{{ template "basic.message" . }}` to indicate that we want to use the message from the "basic" template.

Testing with the "Test" button resulted in the error: `Post \"\": unsupported protocol scheme \"\""`. The logs show only these inscrutable messages and the same error:

```
t=2022-01-08T10:41:29+0000 lvl=info msg="level=error msg=\"Missing receiver\"" logger=alerting.notifier.discord
t=2022-01-08T10:41:29+0000 lvl=info msg="level=error msg=\"Missing group labels\"" logger=alerting.notifier.discord
t=2022-01-08T10:41:29+0000 lvl=eror msg="Failed to send notification to Discord" logger=alerting.notifier.discord error="Post \"\": unsupported protocol scheme \"\""
```

**What you expected to happen**:

This alert template should work correctly, a message should be sent to the target contact point in this format.

**How to reproduce it (as minimally and precisely as possible)**:

Define the alert template as described above, or copy [the example template from the unified alerting documentation](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/message-templating/#custom-template-examples) and name it "basic". Configure a contact point (we used Discord) to use this template by setting the message content to `{{ template "basic.message" . }}`. Test the alert notification using the "Test" button.

**Anything else we need to know?**:

The fix we found was to change the name of `"message"` at its definition in the template to `"basic.message"`. Now the template looks like this:

```
{{ define "alert" }}
  {{ .Annotations.summary }} 
{{ end }}

{{ define "basic.message" }}
  {{ if gt (len .Alerts.Firing) 0 }}
    {{ len .Alerts.Firing }} firing:
    {{ range .Alerts.Firing }} {{ template "alert" .}} {{ end }}
  {{ end }}
{{ end }}
```

We noticed that in the default alert template, [the message is indeed defined as "default.message" rather than just "message"](https://github.com/grafana/grafana/blob/main/pkg/services/ngalert/notifier/channels/default_template.go#L31). So it also works by restating its name in the message section. The documentation on alert templating implies this shouldn't be necessary though, because [its examples use `{{ define "message" }}`](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/message-templating/#custom-template-examples). This exact example, copied and pasted into Grafana, doesn't work either.

This is an implementation bug rather than a documentation bug IMHO, because we shouldn't have to restate the template name in the contents of the template.

**Environment**:
- Grafana version: v8.3.3 (30bb7a93ca)
- Data source: prometheus 2.32.1
- OS installed on: Docker image on debian buster (arm64)
- User OS: Chrome on Win10
- Grafana plugins: default with docker image

**Comment by gerobinson** [comment]
Hi,

The error message `Post \"\": unsupported protocol scheme \"\""` makes me think the `Webhook URL` for the contact point does not have `https://` at the start of the URL. I can't see how the template would cause this error message. Would it be possible to check the configuration for the contact point?

**Comment by gerobinson** [comment]
Hi @shiinee, is this issue resolved?

**Comment by GiovanniBattista** [comment]
Hi, I'm having the same issue as @shiinee when using a custom message template. I can confirm that the Webhook URL for discord does start with https. It is also working with the default message template, but not with a custom one.
![grafik](https://user-images.githubusercontent.com/12544460/152134354-f3919eab-5a09-4952-8ca6-5ebcf53c2829.png)

**Comment by gerobinson** [comment]
Hi @GiovanniBattista, thanks for the screenshot! I have found the issue and opened a fix.

**Comment by gerobinson** [comment]
Hi, the fix for this issue will be released in the next minor release 8.3.6.
